homeassistant/components/switch/tuya.py

```diff
@@ -0,0 +1,55 @@
+"""
+Support for Tuya switch.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.tuya/
+"""
+import asyncio
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.components.tuya import DOMAIN, DATA_TUYA, TuyaDevice
+
+DEPENDENCIES = ['tuya']
+
+DEVICE_TYPE = 'switch'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Tuya Switch device."""
+    tuya = hass.data[DATA_TUYA]
+    devices = tuya.get_devices_by_type(DEVICE_TYPE)
+
+    if DEVICE_TYPE not in hass.data[DOMAIN]['entities']:
+        hass.data[DOMAIN]['entities'][DEVICE_TYPE] = []
+
+    for device in devices:
+        if device.object_id() not in hass.data[DOMAIN]['dev_ids']:
+            add_devices([TuyaSwitch(device, hass)])
+            hass.data[DOMAIN]['dev_ids'].append(device.object_id())
+
+
+class TuyaSwitch(TuyaDevice, SwitchDevice):
+    """Tuya Switch Device."""
+
+    def __init__(self, tuya, hass):
+        """Init Tuya switch device."""
+        super(TuyaSwitch, self).__init__(tuya, hass)
+        self.entity_id = DEVICE_TYPE + '.' + tuya.object_id()
+
+    @asyncio.coroutine
```

Comment:
We now use the async syntax of Python 3.5. So remove this decorator and define the method with `async def`.